### ISSUE TITLE
Import Schematic: "Index was outside the bounds of the array."

### ISSUE THREAD

**Issue by GregariousJB** [TEdit/Terraria-Map-Editor#1264]
**Describe the bug**
Some schematics import fine (even larger, older ones from 2016). But when importing certain specific schematics, the error "Index was outside the bounds of the array." appears. It seems to have something to do with what it's trying to import. Maybe some wiring/timers or active stone blocks? Not sure.

**To Reproduce**
1. Go to 'Clipboard' section in the right sidebar of TEdit
2. Click on 'Import Schematic or Image'
3. Load specific schematic

**Expected behavior**
The schematic should load, appearing in the gray space below the Clipboard section.

**Screenshots**
![image](https://i.imgur.com/k2s4pqX.png)

**Desktop (please complete the following information):**
 - OS: Windows 10 64bit LTSC 1809
 - Version Tedit 4 beta 12
 - Terraria Version 1.4.0.4

**Additional context**
Here's download links to several schematics having issues:

https://drive.google.com/file/d/1GOhf3lPbsYLjE7rv5WFgzcAerbqRdf5S/view?usp=sharing
https://drive.google.com/file/d/1x-6GT164gb2w3Kn3pkYDloOG07IFExHH/view?usp=sharing
https://drive.google.com/file/d/18hGh0ifld035R3Oeyh12qm2C9Qb2qK8b/view?usp=sharing

Thanks in advance.

**Comment by BinaryConstruct** [comment]
Thank you for including the files, they will help immensely.

**Comment by BinaryConstruct** [comment]
I think this has to do with some of the tile types being framed now. I ran through quite a few debugging scenarios and was unable to get these to work.

My suggestion is to use the 1.3 version of TEdit, create a new world, paste in these schematics. Open that world in terraria to upgrade it, then load that world in TEdit 4. You should be able to recreate the schematics this way.

**Comment by GregariousJB** [comment]
Decided to try your suggestion finally, but I'm getting the same error using [TEdit v3.10.18105.218 (for Terraria 1.3.5)](https://github.com/TEdit/Terraria-Map-Editor/releases/tag/3.10.18105.218).

My steps:
1. Open TEdit 3.10 (or 3.11 as it says on the Releases page on Github? I'm confused)
2. File > New
3. Click OK to get a default, empty world
4. Clipboard > Import Schematic
5. Opening arena_jungle and arena_hallow (file downloads above) will result in this error. 

Oddly, arena_crimson (the 3rd file in my list above) now works in TEdit 4.0 Beta 27, but an almost perfect split down the center and the entire right half of the schematic is moved down about 5 blocks. Selecting the right half and copy/pasting it back to where it belonged was an easy fix. Strange.

It's not the end of the world for me. These are mostly just simple arena schematics, I just thought you should know in case there's other people having this issue with importing schematics, to try and help you narrow it down. I suspect the schematics are just corrupted somehow. Thanks again.

**Comment by BinaryConstruct** [comment]
I spent a few days looking at the files and testing the schematic functionality when you first reported this. I made some improvements to the loader for old schematics and drastic improvements to the new format. Unfortunately, I was never able to get these particular files to work properly. I suspect they were made with a specific version of TEdit, and something changed in the later version with tile formats. Without being able to track down exactly what changed I would have to try every release for 1.3.5. the new format tracks what version of TEdit created it, so at least it shouldn't be a problem for future schematics.
